include/onnxruntime/core/framework/kernel_registry.h

```diff
@@ -29,10 +29,16 @@ class KernelRegistry {
                          const DataTransferManager& data_transfer_mgr,
                          std::unique_ptr<OpKernel>& op_kernel) const ORT_MUST_USE_RESULT;
 
-  // Check if an execution provider can create kernel for a node and return
-  // the kernel if so
-  const KernelCreateInfo* TryFindKernel(const onnxruntime::Node& node,
-                                        onnxruntime::ProviderType exec_provider) const;
+  // Check if an execution provider can create kernel for a node and return the kernel if so
+  Status TryFindKernel(const onnxruntime::Node& node,
+                       onnxruntime::ProviderType exec_provider, const KernelCreateInfo** out) const;
+
+  bool HasImplementationOf(const onnxruntime::Node& node,
```

Comment:
nit: This function doesn't need to be a method of this class. It doesn't need access to any private data of this class. Making this a non-member yields greater encapsulation. A static method of this class is probably better or in a separate class called KernelRegistryUtils (although the latter would be overkill for just one method).

Reply:
Sorry, which one? HasImplementationOf? It calls TryFindKernel.

Reply:
Will do it. Thank you!